File: 2015/js/day1.js
const map = Deno.readTextFileSync("./input/day1.txt").trimEnd().split("");

// part 1

let floor = 0;

for (let i = 0; i < map.length; i++) {
  if (map[i] === "(") {
    floor++;
  } else if (map[i] === ")") {
    floor--;
  }
}

console.log("The map leads to floor: ", floor);

//part part 2

floor = 0;
let step = 0;

for (let i = 0; i < map.length; i++) {
  if (map[i] === "(") {
    floor++;
  } else if (map[i] === ")") {
    floor--;
    if (floor === -1) {
      step = i + 1;
      break;
    }
  }
}

console.log("step that Santa first goes into the basement is step: ", step);

File: 2015/py/day1.py
with open('../js/input/day1.txt') as txtFile:
    input = txtFile.read()

print(input)
count = 0
basement = 0

for i, ch in enumerate(input):
    if ch == "(" :
        count += 1
    elif ch == ")":
        count -= 1
    if basement == 0 and count == -1:
       basement = i + 1 
        
       
print ("total characters", len(input))
print ("the total amount of presents delivered", count)
print ("Santa goes to basement on step", basement)
